utils/session/with-verify-session-api-route.ts

```diff
@@ -0,0 +1,28 @@
+import { withIronSessionApiRoute } from 'iron-session/next';
+import { NextApiHandler, NextApiRequest, NextApiResponse } from 'next';
+import { hasSessionId, sessionOptions } from '.';
+
+/**
+ * withVerifySessionApiRoute
+ *
+ * @description
+ * This function is a wrapper for API routes that
+ * verifies if a session is started before executing
+ * the handler.
+ *
+ * @param handler
+ * @returns
+ */
+
+export function withVerifySessionApiRoute(handler: NextApiHandler) {
+  function verifySession(req: NextApiRequest, res: NextApiResponse) {
+    if (!hasSessionId(req.session)) {
+      res.status(401).json({ message: 'Session not started' });
```

Comment:
Error message might be too explicit ? I fear a scrapper would immediately know what to look for.

Also I would add a warning in Sentry